From: Jan Engelhardt <jengelh@inai.de>
Date: 2012-12-06 13:56:32.282683397 +0100

build: resolve link failure

qsolve_main.cpp uses functions from gmpxx, and so must link to it.

	g++ -D__STDC_LIMIT_MACROS -DNDEBUG -fmessage-length=0 -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -o .libs/4ti2gmp 4ti2gmp-circuits_main.o 4ti2gmp-groebner_main.o 4ti2gmp-markov_main.o 4ti2gmp-minimize_main.o 4ti2gmp-normalform_main.o 4ti2gmp-qsolve_main.o 4ti2gmp-rays_main.o 4ti2gmp-walk_main.o 4ti2gmp-zbasis_main.o 4ti2gmp-main.o  ./.libs/lib4ti2gmp.so -lglpk  -Wl,--rpath -Wl,/usr/lib64
	/usr/lib64/gcc/x86_64-suse-linux/4.7/../../../../x86_64-suse-linux/bin/ld: 4ti2gmp-qsolve_main.o: undefined reference to symbol '_ZlsRSoPK12__mpz_struct'
	/usr/lib64/gcc/x86_64-suse-linux/4.7/../../../../x86_64-suse-linux/bin/ld: note: '_ZlsRSoPK12__mpz_struct' is defined in DSO /usr/lib64/libgmpxx.so.4 so try adding it to the linker command line
	/usr/lib64/libgmpxx.so.4: could not read symbols: Invalid operation
---
 src/groebner/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: 4ti2-1.3.2/src/groebner/Makefile.am
===================================================================
--- 4ti2-1.3.2.orig/src/groebner/Makefile.am
+++ 4ti2-1.3.2/src/groebner/Makefile.am
@@ -269,7 +269,7 @@ lib4ti2int64_la_SOURCES = $(lib4ti2sourc
 # Arbitrary precision flags.
 # 4ti2 uses GMP (GLPL), an arbitrary precision arithmetic library.
 if HAVE_GMP_WITH_CXX
-4ti2gmp_LDADD = lib4ti2gmp.la
+4ti2gmp_LDADD = lib4ti2gmp.la -lgmpxx
 4ti2gmp_CPPFLAGS = -D_4ti2_GMP_ $(GMP_CFLAGS)
 4ti2gmp_SOURCES = $(4ti2sources)
 lib4ti2gmp_la_CPPFLAGS = -D_4ti2_GMP_
